docs/docker/README.md

```diff
@@ -0,0 +1,184 @@
+
+[Notes for non-Linux systems](notes-for-non-linux.md)
+
+## Usage instructions ##
+
+### Step 1 - Setup network ###
+Create a network in order for the cBioPortal container and mysql database to communicate.
+```
+docker network create cbio-net
+```
+
+### Step 2 - Run mysql with seed database ###
+Start a MySQL server. The command below stores the database in a folder named
+`/<path_to_save_mysql_db>/db_files/`. This should be an absolute path.
+
+```
+docker run -d --restart=always \
+  --name=cbioDB \
+  --net=cbio-net \
+  -e MYSQL_ROOT_PASSWORD='P@ssword1' \
+  -e MYSQL_USER=cbio \
+  -e MYSQL_PASSWORD='P@ssword1' \
+  -e MYSQL_DATABASE=cbioportal \
+  -v /<path_to_save_mysql_db>/db_files/:/var/lib/mysql/ \
+  mysql:5.7
+```
+
+Download the seed database from the
+[cBioPortal Datahub](https://github.com/cBioPortal/datahub/blob/master/seedDB/README.md),
+and use the command below to upload the seed data to the server started above.
+
+Make sure to replace
+`/<path_to_seed_database>/seed-cbioportal_<genome_build>_<seed_version>.sql.gz`
+with the path and name of the downloaded seed database. Again, this should be
+an absolute path.
+
+```
+docker run \
+  --name=load-seeddb \
+  --net=cbio-net \
+  -e MYSQL_USER=cbio \
+  -e MYSQL_PASSWORD='P@ssword1' \
+  -v /<path_to_seed_database>/cgds.sql:/mnt/cgds.sql:ro \
+  -v /<path_to_seed_database>/seed-cbioportal_<genome_build>_<seed_version>.sql.gz:/mnt/seed.sql.gz:ro \
+  mysql:5.7 \
+  sh -c 'cat /mnt/cgds.sql | mysql -hcbioDB -u"$MYSQL_USER" -p"$MYSQL_PASSWORD" cbioportal \
+      && zcat /mnt/seed.sql.gz |  mysql -hcbioDB -u"$MYSQL_USER" -p"$MYSQL_PASSWORD" cbioportal'
+```
+
+Follow the logs of this step to ensure that no errors occur. If any error
+occurs, make sure to check it. A common cause is pointing the `-v` parameters
+above to folders or files that do not exist.
+
+### Step 3 - Set up a portal.properties file ###
+
+Copy the
+[`portal.properties.EXAMPLE`](../../src/main/resources/portal.properties.EXAMPLE)
+and change it according to your wishes. See the [full
+reference](portal.properties-Reference.md) and the [skin
+properties](Customizing-your-instance-of-cBioPortal.md) for more information on
+the relevant properties.
+
+Make sure to at least provide the database parameters from step 1, which are
+required for the next step:
+
+```
+db.user=cbio
+db.password=P@ssword1
+db.host=cbioDB
+db.portal_db_name=cbioportal
+db.connection_string=jdbc:mysql://cbioDB/
+```
+
+### Step 4 - Migrate database to latest version ###
+
+Update the seeded database schema to match the cBioPortal version
+in the image, by running the following command. Note that this will
+most likely make your database irreversibly incompatible with older
+versions of the portal code.
+
+```
+docker run --rm -it --net cbio-net \
+    -v /<path_to_config_file>/portal.properties:/cbioportal/portal.properties:ro \
+    cbioportal:update-docker-docs \
```

Comment:
will change this tag once approved - after updating the docker image

Reply:
is it possible to have a docker compose doing the following three tasks?

- set up mysql server
- restore mysql from seed database
- migrate to the latest schema

Reply:
I don't even think that that automation requires docker-compose. It's part of the entrypoint script that the MySQL container itself runs before opening the MySQL service port.

Reply:
However, in [thehyve/cbioportal-docker](https://github.com/thehyve/cbioportal-docker#step-2---run-mysql-with-seed-database), we specifically decided that it was more user-friendly to document loading the seed as an explicit separate step.

Back when we did use the entrypoint script to load the seed database in the same command that starts the service, new users were repeatedly confused about when it would or would not load the seed database; it only does so if you mount a `/var/lib/mysql` volume that is completely empty, not if you mount, say, an unsuccessfully loaded database folder or an old database folder with no studies loaded.

Reply:
The migration script is an even more complicated problem, as [the migration process](https://github.com/cBioPortal/cbioportal/blob/v2.2.0/docs/Updating-your-cBioPortal-installation.md#running-the-migration-script) skips blocks of statements from `migration.sql` depending on which version of the schema was already present in the seed database downloaded from Datahub.

Reply:
I am not using migration python script but migration sql statements when running mysql docker container. I extract the portion of sql statements needed to get the schema updated to the latest from the migration.sql file to a sql file. 
Eventually, I will use an open source database migration tool called **_Flyway_** to automate this process.

Reply:
I'd suggest making a pull request (or issue report) to review and agree such things between institutes.

Reply:
@khzhu I think a docker compose file might be useful at some point, especially for developers and testing/CI to have a one command way to run the portal. But yes let's address that later in a separate PR